src/ring_theory/noetherian.lean

```diff
@@ -199,6 +200,45 @@ begin
     { exact zero_smul _ }, { exact λ _ _ _, add_smul _ _ _ } }
 end
 
+lemma singleton_span_is_compact_element (x : M) :
+  complete_lattice.is_compact_element (span R {x} : submodule R M) :=
+begin
+  rw complete_lattice.is_compact_element_iff_le_of_directed_Sup_le,
+  intros d hemp hdir hsup,
+  have : x ∈ Sup d, from (le_def.mp hsup) (mem_span_singleton_self x),
+  obtain ⟨y, ⟨hyd, hxy⟩⟩ := (mem_Sup_of_directed hemp hdir).mp this,
+  exact ⟨y, ⟨hyd, by simpa only [span_le, singleton_subset_iff]⟩⟩,
+end
+
+/-- Finitely generated submodules are precisely compact elements in the submodule lattice -/
+theorem fg_iff_compact (s : submodule R M) : s.fg ↔ complete_lattice.is_compact_element s :=
+begin
+  classical,
+  -- Introduce shorthand for span of an element
+  let sp : M → submodule R M := λ a, span R {a},
+  split,
+  { rintro ⟨t, rfl⟩,
+    rw span_eq_supr_of_singleton_spans,
+    change complete_lattice.is_compact_element (⨆ (x : M) (H : x ∈ t), span R {x}),
```

Comment:
For whatever reason, this line can be replaced with `simp`, but when I run `squeeze_simp` and get `simp only [finset.mem_coe]`, the latter does not work.

Reply:
Yeah, that's a topic [here](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/squeeze_simp.20oddity)

Reply:
If it goes in both directions, maybe we can use `transitivity` to at least make it just one definitional change?

Reply:
Hopefully, once we bump mathlib to lean 3.25.0, you will be able to write `simp?` and get a `simp only` that does work... It would be very cool to test it on problems like this.